oak-core/src/main/java/org/apache/jackrabbit/oak/query/SQL2Parser.java

```diff
@@ -234,6 +249,28 @@ public Query parse(final String query, final boolean initialise) throws ParseExc
 
         return q;
     }
+
+    /**
+     * Checks the InsecureQueryOptionsMode for the values ALLOW or DENY. If allowed, the method returns true,
+     * indicating that insecure options should be returned by the parser. If denied, a ParseException is thrown.
+     * For all other modes, the method returns false, indicating that this option should be ignored.
+     *
+     * @return true if the mode is ALLOW, false otherwise
+     * @throws ParseException if the mode is DENY
+     */
+    private boolean allowsInsecureOptions() throws ParseException {
```

Comment:
I think checking (and throwing the exception) need to happen at execution time, not while parsing.

Reply:
I moved the evaluation to QueryImpl as a distinct `verifyInsecureOptions()` method and call it within the `QueryEngineImpl.executeQuery()` method on each query after it has been parsed/inited but before it has been prepared.